### PR TITLE
Milestone View

### PR DESCRIPTION
**Pull request by mattvv**
The concept is to have a 'Milestone View' where we can use for estimating projects. Crazily enough because this is just a different organization of the way we view tickets, it shouldn't require too many UI changes.

The concept for this view is when we're at the very beginning of a project, we would like to group different tickets by milestones (which end up being labels in pivotal).

For now I've made the milestones static for our needs, but a future PR could probably have these be dynamic. You'll see in this PR this is almost exclusively state changes, and very little visual, the power of redux!

Todos:
- [X] Create a list of stories ids by milestone
- [X] Update list of milestones when pulling down stories
- [X] Add a 'mode' for which view we are viewing the board in, stories or milestones
- [X] Update Add new story to respect mode
- [X] Make sure we can cancel when adding a new story
- [X] Saving a new story should save to both first phases of each mode.
- [X] Dragging and dropping should add Tags for correct mode in Milestone View

### CODE REVIEW

File: pages/projects/[id].tsx [hunk 1]
```diff
@@ -116,6 +121,46 @@ const Project: NextPage = (): JSX.Element => {
       return;
     }
 
+    const landingIndex =
```
Comment (mattvv) [inline pages/projects/[id].tsx:123]: I need to extract this whole method out into some kind of saga/action, (i think we need to do it for all), but we'll see.

File: pages/projects/[id].tsx [hunk 2]
```diff
@@ -190,7 +228,7 @@ const Project: NextPage = (): JSX.Element => {
         )}
         {!loading && (
           <DragDropContext onDragEnd={onDragEnd}>
-            {STORY_STATES.map((state: string, idx: number) => (
+            {selectedModeStates.map((state: string, idx: number) => (
```
Comment (mattvv) [inline pages/projects/[id].tsx:231]: Allows display of whichever mode we're in's stories

File: redux/reducers/stories/byProject.ts [hunk 3]
```diff
@@ -1,56 +1,94 @@
...
   SAVED_NEW_STORY,
+  TOGGLE_MODE,
 } from '../../actions/stories.actions';
-import { Story } from '../../types';
+import { Story, StoryModes } from '../../types';
 import { StoriesByProject } from '../../types';
 
 const initialState = {};
 
+const getStoryMilestone = (story: Story): string => {
```
Comment (mattvv) [inline redux/reducers/stories/byProject.ts:23]: quick and dirty get the story from the list of labels

File: redux/selectors/stories.selectors.ts [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
 
 export const getSelectedProjectId = (state: State): string => {
   return state.stories.selectedProjectId;
 };
 
+export const getSelectedProjectMode = (state: State): string => {
+  return state.stories.byProject[state.stories.selectedProjectId]?.selectedMode;
+};
+
 export const filterStories = (
```
Comment (mattvv) [inline redux/selectors/stories.selectors.ts:11]: will now return stories based also on what mode you're in

File: redux/reducers/stories/byProject.ts [hunk 5]
```diff
@@ -1,56 +1,94 @@
...
+
+  return 'backlog';
+};
+
 const reducer = (state: Record<string, StoriesByProject> = initialState, action: AnyAction) => {
   switch (action.type) {
     case NEW_STORY: {
+      const selectedMode = state[action.projectId].selectedMode;
+      const idArrayName = selectedMode === 'Milestone' ? 'storyIdsByMilestone' : 'storyIdsByState';
+      const startingState = selectedMode === 'Milestone' ? STORY_MILESTONES[0] : STORY_STATES[0];
```
Comment (jansen0825) [inline redux/reducers/stories/byProject.ts:40]: (nit)
```suggestion
      const isMilestoneMode = state[action.projectId].selectedMode === 'Milestone';
      const idArrayName = isMilestoneMode ? 'storyIdsByMilestone' : 'storyIdsByState';
      const startingState = isMilestoneMode ? STORY_MILESTONES[0] : STORY_STATES[0];
```

**Review by jansen0825 — APPROVED** [review]
Very clean implementation! 💪🏼

File: redux/reducers/stories/byProject.ts (hunk 5 above)
Comment (mattvv) [reply redux/reducers/stories/byProject.ts:40]: Yeah this name is nicer

**Review by kazy-kode — COMMENTED** [review]
lgtm :boom: :rocket:

**Pull request merged**